src/domain/settings/currencies/components/store-currencies/add-currencies-screen.tsx

```diff
@@ -0,0 +1,153 @@
+import { Currency } from "@medusajs/medusa"
+import { useAdminCurrencies, useAdminUpdateStore } from "medusa-react"
+import React, { useContext, useState } from "react"
+import { usePagination, useRowSelect, useSortBy, useTable } from "react-table"
+import Button from "../../../../../components/fundamentals/button"
+import Modal from "../../../../../components/molecules/modal"
+import { LayeredModalContext } from "../../../../../components/molecules/modal/layered-modal"
+import useNotification from "../../../../../hooks/use-notification"
+import { getErrorMessage } from "../../../../../utils/error-messages"
+import { useEditCurrenciesModal } from "./edit-currencies-modal"
+import CurrenciesTable from "./table"
+import { useCurrencyColumns } from "./use-currency-table-columns"
+
+const LIMIT = 15
+
+const AddCurrenciesScreen = () => {
+  const [offset, setOffset] = useState(0)
+  const [selectedRowIds, setSelectedRowIds] = useState<string[]>([])
+
+  const { onClose, store } = useEditCurrenciesModal()
+  const { reset, pop } = useContext(LayeredModalContext)
+
+  const { currencies, count, status } = useAdminCurrencies(
+    {
+      limit: LIMIT,
+      offset,
+    },
+    {
+      keepPreviousData: true,
+    }
+  )
+
+  const { mutate } = useAdminUpdateStore()
+  const notification = useNotification()
+
+  const onSubmit = (next: () => void) => {
+    mutate(
+      {
+        currencies: [
+          ...store.currencies.map((curr) => curr.code),
+          ...selectedRowIds,
+        ],
+      },
+      {
+        onSuccess: () => {
+          notification("Success", "Successfully updated currencies", "success")
+          next()
+        },
+        onError: (err) => {
+          notification("Error", getErrorMessage(err), "error")
+        },
+      }
+    )
+  }
+
+  const filteredData = React.useMemo(() => {
+    const codes = store.currencies.map((curr) => curr.code) || []
+    return currencies?.filter(({ code }) => !codes.includes(code)) || []
+  }, [currencies, store])
+
+  const columns = useCurrencyColumns()
+
+  const tableState = useTable<Currency>(
+    {
+      data: filteredData || [],
+      columns,
+      manualPagination: true,
+      initialState: {
+        pageIndex: Math.floor(offset / LIMIT),
+        pageSize: LIMIT,
+      },
+      autoResetPage: false,
+      autoResetSelectedRows: false,
+      getRowId: (row) => row.code,
+      pageCount: Math.ceil((count || 0) / LIMIT),
+      defaultColumn: {
+        width: "auto",
+      },
+    },
+    useSortBy,
+    usePagination,
+    useRowSelect
+  )
+
+  if (status === "error") {
+    return <div>Failed to load</div>
+  }
+
+  if (status === "loading" && !currencies) {
+    return <div>Loading...</div>
+  }
```

Comment:
suggestion: can we add a bit of spacing around these messages
![Screenshot 2022-09-15 at 10 58 35](https://user-images.githubusercontent.com/16856471/190368315-c1740e5c-56bd-4908-be06-82cc96eccc71.png)